pkg/controller/clusterHandler.go

```diff
@@ -0,0 +1,213 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// NewClusterHandler initializes the ClusterHandler with the required structures for each cluster.
+func NewClusterHandler() *ClusterHandler {
+	handler := &ClusterHandler{
+		ClusterConfigs:      make(map[string]*ClusterConfig),
+		uniqueAppIdentifier: make(map[string]struct{}),
+		eventQueue:          workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+	}
+	return handler
+}
+
+// fetch cluster name for given secret if it holds kubeconfig of the cluster.
+func (ch *ClusterHandler) getClusterForSecret(name, namespace string) ClusterDetails {
+	ch.RLock()
+	defer ch.RUnlock()
+	for _, mcc := range ch.ClusterConfigs {
+		// Skip empty/nil configs processing
+		if mcc.clusterDetails == (ClusterDetails{}) {
+			continue
+		}
+		// Check if the secret holds the kubeconfig for a cluster by checking if it's referred in the multicluster config
+		// if so then return the cluster name associated with the secret
+		if mcc.clusterDetails.Secret == (namespace + "/" + name) {
+			return mcc.clusterDetails
+		}
+	}
+	return ClusterDetails{}
+}
+
+// addClusterConfig adds a new cluster configuration to the ClusterHandler.
+func (ch *ClusterHandler) addClusterConfig(clusterName string, config *ClusterConfig) {
+	ch.Lock()
+	ch.ClusterConfigs[clusterName] = config
+	ch.Unlock()
+}
+
+// deleteClusterConfig removes a cluster configuration from the ClusterHandler.
+func (ch *ClusterHandler) deleteClusterConfig(clusterName string) {
+	ch.Lock()
+	delete(ch.ClusterConfigs, clusterName)
+	ch.Unlock()
+}
+
+// getClusterConfig returns the cluster configuration for the specified cluster.
+func (ch *ClusterHandler) getClusterConfig(clusterName string) *ClusterConfig {
+	ch.RLock()
+	defer ch.RUnlock()
+	if _, exists := ch.ClusterConfigs[clusterName]; !exists {
+		return nil
+	}
+	return ch.ClusterConfigs[clusterName]
+}
+
+// addInformerStore adds a new InformerStore to the ClusterHandler.
+func (ch *ClusterHandler) addInformerStore(clusterName string, store *InformerStore) {
+	ch.Lock()
+	if clusterConfig, ok := ch.ClusterConfigs[clusterName]; !ok {
+		clusterConfig.InformerStore = store
+	}
+	ch.Unlock()
+}
+
+// getInformerStore returns the InformerStore for the specified cluster.
+func (ch *ClusterHandler) getInformerStore(clusterName string) *InformerStore {
+	ch.RLock()
+	defer ch.RUnlock()
+	if clusterConfig, exists := ch.ClusterConfigs[clusterName]; exists {
+		return clusterConfig.InformerStore
+	}
+	return nil
+}
+
+// enqueueEvent adds an event to the eventQueue after checking for uniqueness.
+func (ch *ClusterHandler) enqueueEvent(clusterName string, obj interface{}) {
+	key := fmt.Sprintf("%s/%s", clusterName, "")
+	if _, exists := ch.uniqueAppIdentifier[key]; exists {
+		fmt.Printf("Duplicate event discarded: %s\n", key)
+		return
+	}
+	ch.uniqueAppIdentifier[key] = struct{}{}
+	ch.eventQueue.Add(obj)
+	fmt.Printf("Event queued: %s\n", key)
```

Comment:
Is it good to add the logger functions instead of `fmt` ?

Reply:
This function is a placeholder for now.

Reply:
OK